server/lib/src/idm/oauth2.rs

```diff
@@ -1873,15 +1878,31 @@ impl IdmServerProxyReadTransaction<'_> {
             admin_warn!("Unsupported OAuth2 response_type (should be 'code')");
             return Err(Oauth2Error::UnsupportedResponseType);
         }
+
         let Some(response_mode) = auth_req.get_response_mode() else {
-            admin_warn!(
+            warn!(
                 "Invalid response_mode {:?} for response_type {:?}",
-                auth_req.response_mode,
-                auth_req.response_type
+                auth_req.response_mode, auth_req.response_type
             );
             return Err(Oauth2Error::InvalidRequest);
         };
 
+        let response_mode = match response_mode {
+            ResponseMode::Query => SupportedResponseMode::Query,
+            ResponseMode::Fragment => SupportedResponseMode::Fragment,
+            ResponseMode::FormPost => {
+                warn!(
+                    "Invalid response mode form_post requested - many clients request this incorrectly but proceed with response_mode=query. Remapping to query."
+                );
+                warn!("This behaviour WILL BE REMOVED in a future release.");
```

Comment:
sorry to necro but this is gonna be a hard sell. Once we support it removing support for it becomes significantly harder given that we would be breaking people's existing working setups.

Reply:
We've done it before - before we remove it, we can convert it to a compatibility boolean, and then gauge affected users.